src/Sylius/Bundle/PayumBundle/Form/Type/PaypalGatewayConfigurationType.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\PayumBundle\Form\Type;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\CallbackTransformer;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+final class PaypalGatewayConfigurationType extends AbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('username', TextType::class, [
+                'label' => 'sylius.form.gateway_configuration.paypal.username',
+                'constraints' => [
+                    new NotBlank([
+                        'message' => 'sylius.gateway_config.paypal.username.not_blank',
+                        'groups' => 'sylius',
+                    ])
+                ],
+            ])
+            ->add('password', PasswordType::class, [
+                'label' => 'sylius.form.gateway_configuration.paypal.password',
+                'constraints' => [
+                    new NotBlank([
+                        'message' => 'sylius.gateway_config.paypal.password.not_blank',
+                        'groups' => 'sylius',
+                    ])
+                ],
+            ])
+            ->add('signature', TextType::class, [
+                'label' => 'sylius.form.gateway_configuration.paypal.signature',
+                'constraints' => [
+                    new NotBlank([
+                        'message' => 'sylius.gateway_config.paypal.signature.not_blank',
+                        'groups' => 'sylius',
+                    ])
+                ],
+            ])
+            ->add('payum_http_client', HiddenType::class, [
```

Comment:
It would be good but it would require separate factory method and therefore separate route (which is too smelly) or merging default values with passed in factory method (which is too tricky). I think it could be left as it is for one more reason - in the future we could (and we probably should) think about supporting custom payum clients services - and it would only require to change field type from hidden to choice.